src/mlpack/methods/decision_tree/multiple_random_dimension_select.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file multiple_random_dimension_select.hpp
+ * @author Ryan Curtin
+ *
+ * Select a number of random dimensions to pick from.
+ */
+#ifndef MLPACK_METHODS_DECISION_TREE_MULTIPLE_RANDOM_DIMENSION_SPLIT_HPP
+#define MLPACK_METHODS_DECISION_TREE_MULTIPLE_RANDOM_DIMENSION_SPLIT_HPP
+
+namespace mlpack {
+namespace tree {
+
+/**
+ * This dimension selection policy allows the selection from a few random
+ * dimensions.  The number of random dimensions to use is specified by the
+ * NumDimensions template parameter.
+ *
+ * @tparam NumDimensions Number of random dimensions to select.
+ */
+template<size_t NumDimensions = 3>
+class MultipleRandomDimensionSelect
+{
+ public:
+  /**
+   * Instantiate the MultipleRandomDimensionSelect object.
+   */
+  MultipleRandomDimensionSelect(const size_t dimensions)
+  {
+    for (size_t i = 0; i < NumDimensions; ++i)
+    {
+      // Generate random different numbers.
+      bool unique = false;
+      size_t value;
+      while (!unique)
+      {
+        value = math::RandInt(dimensions);
+
+        // Check if we already have the value.
+        unique = true;
+        for (size_t j = 0; j < i; ++j)
+        {
+          if (values[j] == value)
+          {
+            unique = false;
+            break;
+          }
+        }
+      }
+
+      values[i] = value;
+    }
+
+    values[NumDimensions] = size_t(-1);
```

Comment:
I thought `size_t(-1)` was `std::numeric_limits<size_t>::max()`, but I'll just go ahead and change it.